smart_open/tests/test_s3.py

```diff
@@ -21,9 +21,13 @@
 import boto3
 import botocore.client
 import botocore.endpoint
-import moto
 import pytest
 
+try:
+    from moto import mock_s3
+except ImportError:
+    from moto import mock_aws as mock_s3
```

Comment:
A code comment would be great (including a hyperlink to the relevant github ticket/PR), to clarify what's going on for our future selves.

And also to remember to simplify the code once "old moto" is sufficiently old = no need to support moto<5 any more.